tests/test_small_output.c

```diff
@@ -0,0 +1,68 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include <librsync.h>
+
+// Test behaviour when the output buffer is too small, and grows bit by bit
+// Parameters: test_small_output.test  INPUTSIZE  INITIAL_BUFFER_SIZE  OUTPUT_GROWTH_RATE
+
+int main(int argc, const char** argv)
+{
+   int i;
+   size_t input_size = atoi(argv[1]);
+   size_t start = atoi(argv[2]);
+   size_t step = atoi(argv[3]);
+
+   printf("Input size: %d\n", input_size);
+   printf("Output buffer start at %d step growth of %d\n", start, step);
+
+   // make this silly-huge, we don't want to test part
+   char * outbuf = (char*)malloc(1024+10*input_size);
+
+   // load with anything
+   char * in = (char*)malloc(input_size);
+   for (i = 0; i < input_size; ++i)
+      in[i] = 'x';
+
+   // generate a signature
+   rs_buffers_t buf;
+   buf.next_out = outbuf;
+   buf.avail_out = start;
+   buf.next_in = (char*)(in);   // TODO why is this not const void* ?
```

Comment:
Moving all the data pointers to `void*` would be a worthwhile separate change, but probably counts as an API break.